core/controllers/skill_editor.py

```diff
@@ -200,9 +201,28 @@ def delete(self, skill_id):
         skill_domain.Skill.require_valid_skill_id(skill_id)
         skill_ids_assigned_to_some_topic = (
             topic_services.get_all_skill_ids_assigned_to_some_topic())
+
         if skill_id in skill_ids_assigned_to_some_topic:
-            raise self.InvalidInputException(
-                'Cannot delete skill that is assigned to a topic.')
+            all_topics = topic_fetchers.get_all_topics()
+            for topic in all_topics:
+                change_list = []
+                if skill_id in topic.get_all_skill_ids():
+                    for subtopic in topic.subtopics:
+                        if skill_id in subtopic.skill_ids:
+                            change_list.append(topic_domain.TopicChange({
+                                'cmd': 'remove_skill_id_from_subtopic',
+                                'subtopic_id': subtopic.id,
+                                'skill_id': skill_id
+                            }))
+
+                    change_list.append(topic_domain.TopicChange({
+                        'cmd': 'remove_uncategorized_skill_id',
+                        'uncategorized_skill_id': skill_id
+                    }))
+                    topic_services.update_topic_and_subtopic_pages(
+                        self.user_id, topic.id, change_list,
+                        'Removed %s from the topic' % skill_id)
```

Comment:
Is this the commit message? If so it might not be super useful, given that the skill ID is a random-looking string of characters -- you might want to use the name instead. (Removed skill "..." from the topic.)

Reply:
Yeah you are right. Done